event-hubs.api.md

```diff
@@ -0,0 +1,287 @@
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    // (undocumented)
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface LastEnqueuedEventInfo {
+    enqueuedTime?: Date;
+    offset?: string;
+    retrievalTime?: Date;
+    sequenceNumber?: number;
+}
+
+export { MessagingError }
+
+// @public
+export interface PartitionCheckpointer {
+    updateCheckpoint(eventData: ReceivedEventData): Promise<void>;
+    updateCheckpoint(sequenceNumber: number, offset: number): Promise<void>;
+    // @internal (undocumented)
+    updateCheckpoint(eventDataOrSequenceNumber: ReceivedEventData | number, offset?: number): Promise<void>;
+}
+
+// @public
+export interface PartitionContext {
+    consumerGroupName: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    partitionId: string;
+}
+
+// @public
+export interface PartitionManager {
+    claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]>;
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    listOwnership(fullyQualifiedNamespace: string, eventHubName: string, consumerGroupName: string): Promise<PartitionOwnership[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export interface PartitionOwnership extends PartitionContext {
+    eTag?: string;
+    lastModifiedTimeInMs?: number;
+    ownerId: string;
+    ownerLevel: number;
+}
+
+// @public
+export interface PartitionProperties {
+    beginningSequenceNumber: number;
+    eventHubName: string;
+    lastEnqueuedOffset: number;
+    lastEnqueuedSequenceNumber: number;
+    lastEnqueuedTimeUtc: Date;
+    partitionId: string;
+}
+
+// @public
+export type ProcessCloseHandler = (reason: CloseReason, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessErrorHandler = (error: Error, context: SubscriptionPartitionContext) => Promise<void>;
+
+// @public
+export type ProcessEvent = (receivedEvent: ReceivedEventData, context: PartitionContext & PartitionCheckpointer) => Promise<void>;
+
+// @public
+export type ProcessInitializeHandler = (context: SubscriptionPartitionContext & SubscriptionPartitionInitializer) => Promise<void>;
+
+// @public
+export interface ReceivedEventData {
+    body: any;
+    enqueuedTimeUtc: Date;
+    offset: number;
+    partitionKey: string | null;
+    properties?: {
+        [key: string]: any;
+    };
+    sequenceNumber: number;
+    systemProperties?: {
+        [key: string]: any;
+    };
+}
+
+export { RetryOptions }
+
+// @public
+export interface SendBatchOptions {
```

Comment:
Would it be confusing to have both `SendBatchOptions` and `CreateBatchOptions`?

Reply:
I think 'no' - the user's only going to see these mentioned on their respective calls (sendBatch() and createBatch()).